src/types.rs

```diff
@@ -1086,6 +1087,129 @@ impl SimpleAsn1Writable for GeneralizedTime {
     }
 }
 
+/// Used for parsing and writing ASN.1 `GENERALIZED TIME` values accepting
+/// fractional seconds value.
+/// See https://github.com/alex/rust-asn1/issues/491 for discussion.
+#[derive(Debug, Clone, PartialEq, PartialOrd, Hash, Eq)]
+pub struct GeneralizedTimeFractional {
+    pub datetime: DateTime,
+    pub fraction: Option<u32>, // Up to 1 ns precision (10^9)
+}
+
+impl GeneralizedTimeFractional {
+    pub fn new(dt: DateTime, fraction: Option<u32>) -> ParseResult<GeneralizedTimeFractional> {
```

Comment:
This needs to enforce the various invariants. Notably that the fractional seconds are not >= 1e9.